apis/python/src/tiledbsoma/tiledb_session_context.py

```diff
@@ -0,0 +1,39 @@
+import os
+from dataclasses import dataclass
+from typing import Optional, cast
+
+import tiledb
+
+
+def build_default_tiledb_ctx() -> tiledb.Ctx:
+    """
+    Build a TileDB context starting with reasonable defaults, and overriding and updating with user-provided config
+    options.
+    """
+
+    # Note: Defaults must provide positive out-of-the-box UX!
+
+    cfg = {"sm.mem.reader.sparse_global_order.ratio_array_data": 0.3}
+
+    # This is necessary for smaller tile capacities when querying with a smaller memory budget.
+
+    # Temp workaround pending https://app.shortcut.com/tiledb-inc/story/23827
+    region = os.getenv("AWS_DEFAULT_REGION")
+    if region is not None:
+        cfg["vfs.s3.region"] = cast(str, region)  # type: ignore
+
+    return tiledb.Ctx(cfg)
+
+
+@dataclass(frozen=True)
```

Comment:
optional: in a previous change I suggested the use of `attrs` instead of `dataclasses` (too lazy to dig it up right now). In this case it would be simply switching from `@dataclass` to `@attrs.define` here (after adding the import). See also [the general case for attrs over dataclasses](https://www.attrs.org/en/stable/why.html#data-classes).

(Other than that, using a dataclass/attrs class here is perfect.)

Reply:
I do remember seeing some discussion of that!  :)  Will change it to `attrs` for consistency.

Reply:
done